### PR TITLE
Update vmsizes.py for newer Azure API endpoint

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

File: utils/azure/vmsizes/vmsizes.py [hunk 1]
```diff
@@ -41,19 +91,26 @@ def get_vmsizes(subscription_id, location, token):
...
+                    value = {
+                        "vcpus": int(extract_value("vCPUs", item["capabilities"])),
+                        "ram": int(
+                            float(extract_value("MemoryGB", item["capabilities"]))
+                            * 1000
+                        ),
+                        "gpus": int(extract_value("GPUs", item["capabilities"]))
+                        if extract_value("gpus", item["capabilities"]) is not None
+                        else 0,
+                    }
```
Comment (cmd-ntrf) [inline utils/azure/vmsizes/vmsizes.py:105]: The value extraction can be simplified by defining a dictionary first:
```python
                    caps = { el["name"]:el["value"] for el in item["capabilities"] }
                    value = {
                        "vcpus": int(caps.get("vCPUsAvailable", caps["vCPUs"])),
                        "ram":   int(float(caps.get("MemoryGB"))* 1000),
                        "gpus":  int(caps.get("GPUs", 0))
                    }
```

This also fixes the case where for some VMs, `vCPUs != vCPUsAvailable`.

File: utils/azure/vmsizes/vmsizes.py [hunk 2]
```diff
@@ -16,20 +16,70 @@
...
+
+def extract_value(key, data):
+    result_value = None
+
+    for item in data:
+        if item["name"].lower() == key.lower():
+            result_value = item["value"]
+            break
+
+    return result_value
```
Comment (cmd-ntrf) [inline utils/azure/vmsizes/vmsizes.py:68]: No longer needed

File: utils/azure/vmsizes/vmsizes.py [hunk 3]
```diff
@@ -16,20 +16,70 @@
...
+
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser(description='Generate a list of Azure VM sizes')
-    parser.add_argument('--subscription_id')
-    parser.add_argument('--token')
+    parser = argparse.ArgumentParser(description="Generate a list of Azure VM sizes")
+    parser.add_argument(
+        "--subscription_id",
+        required=True,
+        help="az account get-access-token --resource https://management.azure.com",
```
Comment (cmd-ntrf) [inline utils/azure/vmsizes/vmsizes.py:76]: Subscription id can be retrieve with `az account show`

**Review by cmd-ntrf — CHANGES_REQUESTED** [review]

**Comment by ocaisa** [comment]
Apologies for not getting back to this!

**Pull request merged**